modules/apps/portal-workflow/portal-workflow-kaleo-runtime-impl/src/main/java/com/liferay/portal/workflow/kaleo/runtime/internal/assignment/MultiLanguageKaleoTaskAssignmentSelector.java

```diff
@@ -81,41 +81,46 @@ public Collection<KaleoTaskAssignment> getKaleoTaskAssignments(
 		return kaleoTaskAssignments;
 	}
 
-	@Reference(
-		cardinality = ReferenceCardinality.MULTIPLE,
-		policy = ReferencePolicy.DYNAMIC,
-		policyOption = ReferencePolicyOption.GREEDY,
-		target = "(scripting.language=*)"
-	)
-	protected void addKaleoTaskAssignmentSelector(
-			KaleoTaskAssignmentSelector kaleoTaskAssignmentSelector,
-			Map<String, Object> properties)
-		throws KaleoDefinitionValidationException {
-
-		String[] scriptingLanguages = _getScriptingLanguages(properties);
-
-		for (String scriptingLanguage : scriptingLanguages) {
-			_kaleoTaskAssignmentSelectors.put(
-				_getKaleoTaskAssignmentSelectKey(
-					scriptingLanguage,
-					ClassUtil.getClassName(kaleoTaskAssignmentSelector)),
-				kaleoTaskAssignmentSelector);
-		}
+	@Activate
+	protected void activate(BundleContext bundleContext) {
+		_serviceTrackerMap = ServiceTrackerMapFactory.openSingleValueMap(
+			bundleContext, KaleoTaskAssignmentSelector.class,
+			"(scripting.language=*)",
```

Comment:
@W2jx We keep using this property here because there are other `KaleoTaskAssignmentSelector` implementations that are identified by property `assignee.class.name`.

It's weird, from the first look, to see the same interface to be 2 types of services, in which one type is a "subset" of the other type.

Could you please review the implementations, and see if they can be separated, or whether a new interface can be introduced to distinguish these 2 types?

cc @tinatian